### PR TITLE
fixed

### PR DESCRIPTION
**Pull request by LizaBalshv**

### CODE REVIEW

File: src/main/java/mate.academy/lib/Injector.java [hunk 1]
```diff
@@ -1,13 +1,76 @@
...
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (NoSuchMethodException | InvocationTargetException
+                 | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException("Can't create a new instance of" + clazz.getName());
```
Comment (Vasul-Pankiv) [inline src/main/java/mate.academy/lib/Injector.java:66]: use one general exception (see common mistakes)

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/main/java/mate.academy/lib/Component.java [hunk 2]
```diff
@@ -1,5 +1,8 @@
 package mate.academy.lib;
 
-public @interface Component {
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (olekskov) [inline src/main/java/mate.academy/lib/Component.java:6]: Do not forget to add @Target to custom annotations.

File: src/main/java/mate.academy/lib/Inject.java [hunk 3]
```diff
@@ -1,5 +1,8 @@
 package mate.academy.lib;
 
-public @interface Inject {
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (olekskov) [inline src/main/java/mate.academy/lib/Inject.java:6]: Do not forget to add @Target to custom annotations.

**Review by olekskov — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**